File: snakes/danielle-clojure/src/battlesnake/geometry.clj
(ns battlesnake.geometry)

(defn coord [x y] {:x x :y y})

(defn direction-adjacent [origin adjacent]
  (cond
    (and (= (inc (:x origin)) (:x adjacent)) (= (:y origin) (:y adjacent))) :right
    (and (= (dec (:x origin)) (:x adjacent)) (= (:y origin) (:y adjacent))) :left
    (and (= (inc (:y origin)) (:y adjacent)) (= (:x origin) (:x adjacent))) :up
    (and (= (dec (:y origin)) (:y adjacent)) (= (:x origin) (:x adjacent))) :down
    :else nil))

File: snakes/danielle-rust-warp/src/geometry.rs
use crate::state::{Coordinate, MoveDirection};

pub fn adjacent_direction(origin: Coordinate, adjacent: Coordinate) -> Option<MoveDirection> {
    if origin.x + 1 == adjacent.x && origin.y == adjacent.y {
        Some(MoveDirection::Right)
    } else if origin.x - 1 == adjacent.x && origin.y == adjacent.y {
        Some(MoveDirection::Left)
    } else if origin.y + 1 == adjacent.y && origin.x == adjacent.x {
        Some(MoveDirection::Up)
    } else if origin.y - 1 == adjacent.y && origin.x == adjacent.x {
        Some(MoveDirection::Down)
    } else {
        None
    }
}
